network_simulator/plot_tool.py

```diff
@@ -0,0 +1,300 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from statistics import *
+
+class PlotTool(object):
+    """
+    Produces graph of different data structures.
+    """
+    def __init__(self):
+        pass
+
+    def gen_count_interval_list(self, lst):
+        """
+        Computes a tuple list of counts based on a timestamp list.
+
+        :param list lst: list of timestamps to be converted.
+        :return: list
+        """
+        curr_count = 0
+        curr_time = GRAPH_WINDOW_SIZE
+        output = []
+        for idx in range(len(lst)):
+            timestamp = lst[idx]
+            if timestamp > curr_time:
+                output.append((curr_time, curr_count))
+                curr_count = 0
+                curr_time += GRAPH_WINDOW_SIZE
+            if timestamp <= curr_time:
+                curr_count += 1
+        # Append the last elements if they did not reach an end of an
+        # interval.
+        if curr_count != 0:
+            output.append((curr_time, curr_count))
+        return output
+
+    def gen_rate_tuple_list(self, tuple_list):
+        """
+        Computes a tuple list of rates based on a (timestamp, data) tuple
+        list.
+
+        :param list tuple_list: list of tuples to be converted.
+        :return: list
+        """
+        curr_count = 0
+        curr_sum = 0
+        curr_time = GRAPH_WINDOW_SIZE
+        output = []
+        for tup_idx in range(len(tuple_list)):
+            timestamp, data = tuple_list[tup_idx]
+            if timestamp > curr_time:
+                if curr_count != 0:
+                    output.append((curr_time, float(curr_sum) / curr_count))
+                    curr_count = 0
+                    curr_sum = 0
+                curr_time += GRAPH_WINDOW_SIZE
+            if timestamp <= curr_time:
+                curr_count += 1
+                curr_sum += data
+        # Append the last elements if they did not reach an end of an
+        # interval.
+        if curr_count != 0:
+            output.append((curr_time, float(curr_sum) / curr_count))
+        return output
+
+    def graph_tuple_list(self, tuple_list=[], scatter=False):
+        """
+        Graphs the content of a tuple list.
+
+        :param list tuple_list: if we have a newer version of the list,
```

Comment:
Not sure what this comment is referring to. Isn't this just the list being plotted?